configtelemetry/type.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package configtelemetry
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"strings"
+)
+
+var ignoreComments = []string{
+	"telemetry: none",
+	"This field is deprecated and must not be used.",
+}
+
+// typeFieldMap will generate map of it's all possible selectors and adds them as a key. The value
+// is the position of the declaration in the type spec.
+func typeFieldMap(fset *token.FileSet, obj *types.TypeName) (map[string]token.Pos, error) {
+	srcFile := fset.PositionFor(obj.Pos(), false).Filename
+	fileNode, err := parser.ParseFile(fset, srcFile, nil, parser.AllErrors|parser.ParseComments)
+	if err != nil {
+		return nil, fmt.Errorf("could parse %s: %w", srcFile, err)
+	}
+
+	var fieldsMap = make(map[string]token.Pos)
```

Comment:
We are actually redundantly allocating here. The `fields` function anyways creates the map. We can remove the allocation from here.